### PR TITLE
[Windows][Installer] Don't consider NT AUTHORITY as a domain.

### PR DESCRIPTION
**Pull request by julien-lebot**
### What does this PR do?
This PR adds code to handle the case where `DDAGENTUSER_NAME="NT AUTHORITY\SYSTEM"` is passed so that it's not considered a domain.

### Motivation
This PR would allow setting `DDAGENTUSER_NAME="NT AUTHORITY\SYSTEM"` without having to specify a dummy `DDAGENTUSER_PASSWORD` to get the installation to succeed.

### Additional Notes
Maybe there are other names we can exclude from the authorized domains based on the [list of reserved words](https://docs.microsoft.com/en-US/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#table-of-reserved-words).

### Possible Drawbacks / Trade-offs
There should not be any conflict with an existing domain name since `NT AUTHORITY` is amongst the [list of reserved words](https://docs.microsoft.com/en-US/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#table-of-reserved-words).

### Describe how to test/QA your changes
Run the installer with `DDAGENTUSER_NAME="NT AUTHORITY\SYSTEM"`, ensure it installs correctly and that the Agent is indeed running as "NT AUTHORITY\SYSTEM".

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by iglendd — APPROVED** [review]
Looks good to me

File: releasenotes/notes/ntsystem_no_pass_warn-aef5f6f5af08c140.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    On the Windows installer command line it's now possible to specify DDAGENTUSER_NAME="NT AUTHORITY\SYSTEM" without having to also specify DDAGENTUSER_PASSWORD.
```
Comment (maycmlee) [inline releasenotes/notes/ntsystem_no_pass_warn-aef5f6f5af08c140.yaml:4]:
```suggestion
    On the Windows installer command line, it's now possible to specify DDAGENTUSER_NAME="NT AUTHORITY\SYSTEM" without having to also specify DDAGENTUSER_PASSWORD.
```

**Review by maycmlee — APPROVED** [review]
Small edit

**Comment by albertvaka** [comment]
When people install the Agent this way, can they upgrade to a newer version of the Agent?

**Comment by albertvaka** [comment]
I remember upgrades were not working at some point when using Local System. Just asking if you tested it, to know if it was still the case. If we truly support installing as Local System, upgrades must work.

**Pull request merged**